lib/toml-rb/keyvalue.rb

```diff
@@ -1,31 +1,39 @@
+require_relative "inline_table"
+
 module TomlRB
   class Keyvalue
-    attr_reader :value, :symbolize_keys
+    attr_reader :dotted_keys, :value, :symbolize_keys
 
-    def initialize(key, value)
-      @key = key
+    def initialize(dotted_keys, value)
+      @dotted_keys = dotted_keys
       @value = value
       @symbolize_keys = false
     end
 
-    def assign(hash, symbolize_keys = false)
+    def assign(hash, fully_defined_keys, symbolize_keys = false)
       @symbolize_keys = symbolize_keys
-      fail ValueOverwriteError.new(key) if hash.key?(key)
-      hash[key] = visit_value @value
-    end
-
-    def visit_inline_table(inline_table)
-      result = {}
-
-      inline_table.value(@symbolize_keys).each do |k, v|
-        result[key k] = visit_value v
+      dotted_keys_str = @dotted_keys.join(".")
+      keys = symbolize_keys ? @dotted_keys.map(&:to_sym) : @dotted_keys
+      update = keys.reverse.inject(visit_value @value) { |k1, k2| { k2 => k1 } }
```

Comment:
This is not very clear, would you mind to add a comment here explaining this?

Reply:
For dotted keys support,  dotted key string array is converted to a nested hash.
The value data is wrapped in the last key, and the previous keys are wrapped in reverse order.

Please refer to this example.
```ruby
["a", "b", "c"].reverse.inject("value") {|k1, k2| { k2 => k1}}
=> {"a"=>{"b"=>{"c"=>"value"}}}
```